packages/langium/src/lsp/formatter.ts

```diff
@@ -74,33 +74,52 @@ export abstract class AbstractFormatter implements Formatter {
         }
     }
 
-    formatDocumentRange(document: LangiumDocument, params: DocumentRangeFormattingParams): MaybePromise<TextEdit[]> {
+    /**
+     * Returns whether a range for a given document is error free, i.e. safe to format
+     *
+     * @param document Document to inspect for lexer & parser errors that may produce an unsafe range
+     * @param range Formatting range to check for safety
+     * @returns Whether the given formatting range does not overlap with or follow any regions with an error
+     */
+    private isFormatRangeErrorFree(document: LangiumDocument, range: Range): boolean {
```

Comment:
We should never declare a method as `private`, as that makes overriding it unnecessarily complicated. Please use `protected`.

Reply:
That's fair, one moment...